Reproducible Failure in org.elasticsearch.search.aggregations.bucket.histogram.DateHistogramAggregatorTests.testTwoBucketsWithDocCountUsesTraditionalCollection

`org.elasticsearch.search.aggregations.bucket.histogram.DateHistogramAggregatorTests.testTwoBucketsWithDocCountUsesTraditionalCollection"` fails for me every time on current master when running:

```
./gradlew ':server:test' --tests "org.elasticsearch.search.aggregations.bucket.histogram.DateHistogramAggregatorTests.testTwoBucketsWithDocCountUsesTraditionalCollection" -Dtests.seed=B0F2BA0C9CFDACD2 -Dtests.locale=sr-BA -Dtests.timezone=America/Thunder_Bay -Druntime.java=17
```

failure:

```
> Task :server:test
WARNING: A terminally deprecated method in java.lang.System has been called
WARNING: System::setSecurityManager has been called by org.gradle.api.internal.tasks.testing.worker.TestWorker (file:/home/brownbear/.gradle/wrapper/dists/gradle-7.4.2-all/9uukhhbclvbegdvsww0j0cr3p/gradle-7.4.2/lib/plugins/gradle-testing-base-7.4.2.jar)
WARNING: Please consider reporting this to the maintainers of org.gradle.api.internal.tasks.testing.worker.TestWorker
WARNING: System::setSecurityManager will be removed in a future release

REPRODUCE WITH: ./gradlew ':server:test' --tests "org.elasticsearch.search.aggregations.bucket.histogram.DateHistogramAggregatorTests.testTwoBucketsWithDocCountUsesTraditionalCollection" -Dtests.seed=B0F2BA0C9CFDACD2 -Dtests.locale=sr-BA -Dtests.timezone=America/Thunder_Bay -Druntime.java=18

org.elasticsearch.search.aggregations.bucket.histogram.DateHistogramAggregatorTests > testTwoBucketsWithDocCountUsesTraditionalCollection FAILED
    java.lang.AssertionError: Expected a map containing
                     d: a map containing
                delegate: "RangeAggregator.NoOverlap"
          delegate_debug: a map containing
                    ranges: <2>
    average_docs_per_range: <5005.0>
                singletons: expected a value greater than <1> but <1> was equal to <1>
            non-singletons: <0>
        at __randomizedtesting.SeedInfo.seed([B0F2BA0C9CFDACD2:C6AC6B1CEA28B223]:0)
        at org.elasticsearch.test.MapMatcher.assertMap(MapMatcher.java:82)
        at org.elasticsearch.test.MapMatcher.assertMap(MapMatcher.java:65)
        at org.elasticsearch.search.aggregations.bucket.histogram.DateHistogramAggregatorTests.lambda$testTwoBucketsWithDocCountUsesTraditionalCollection$81(DateHistogramAggregatorTests.java:991)
        at org.elasticsearch.search.aggregations.AggregatorTestCase.debugTestCase(AggregatorTestCase.java:772)
        at org.elasticsearch.search.aggregations.AggregatorTestCase.lambda$debugTestCase$11(AggregatorTestCase.java:723)
        at org.elasticsearch.search.aggregations.AggregatorTestCase.withIndex(AggregatorTestCase.java:687)
        at org.elasticsearch.search.aggregations.AggregatorTestCase.debugTestCase(AggregatorTestCase.java:723)
        at org.elasticsearch.search.aggregations.bucket.histogram.DateHistogramAggregatorTests.testTwoBucketsWithDocCountUsesTraditionalCollection(DateHistogramAggregatorTests.java:979)


Suite: Test class org.elasticsearch.search.aggregations.bucket.histogram.DateHistogramAggregatorTests
  1> [2022-05-30T05:35:17,978][INFO ][o.e.s.a.b.h.DateHistogramAggregatorTests] [testTwoBucketsWithDocCountUsesTraditionalCollection] before test
  1> [2022-05-30T05:35:19,236][INFO ][o.e.s.a.b.h.DateHistogramAggregatorTests] [testTwoBucketsWithDocCountUsesTraditionalCollection] after test
  2> REPRODUCE WITH: ./gradlew ':server:test' --tests "org.elasticsearch.search.aggregations.bucket.histogram.DateHistogramAggregatorTests.testTwoBucketsWithDocCountUsesTraditionalCollection" -Dtests.seed=B0F2BA0C9CFDACD2 -Dtests.locale=sr-BA -Dtests.timezone=America/Thunder_Bay -Druntime.java=18
  2> java.lang.AssertionError: Expected a map containing
                     d: a map containing
                delegate: "RangeAggregator.NoOverlap"
          delegate_debug: a map containing
                    ranges: <2>
    average_docs_per_range: <5005.0>
                singletons: expected a value greater than <1> but <1> was equal to <1>
            non-singletons: <0>
        at __randomizedtesting.SeedInfo.seed([B0F2BA0C9CFDACD2:C6AC6B1CEA28B223]:0)
        at org.elasticsearch.test.MapMatcher.assertMap(MapMatcher.java:82)
        at org.elasticsearch.test.MapMatcher.assertMap(MapMatcher.java:65)
        at org.elasticsearch.search.aggregations.bucket.histogram.DateHistogramAggregatorTests.lambda$testTwoBucketsWithDocCountUsesTraditionalCollection$81(DateHistogramAggregatorTests.java:991)
        at org.elasticsearch.search.aggregations.AggregatorTestCase.debugTestCase(AggregatorTestCase.java:772)
        at org.elasticsearch.search.aggregations.AggregatorTestCase.lambda$debugTestCase$11(AggregatorTestCase.java:723)
        at org.elasticsearch.search.aggregations.AggregatorTestCase.withIndex(AggregatorTestCase.java:687)
        at org.elasticsearch.search.aggregations.AggregatorTestCase.debugTestCase(AggregatorTestCase.java:723)
        at org.elasticsearch.search.aggregations.bucket.histogram.DateHistogramAggregatorTests.testTwoBucketsWithDocCountUsesTraditionalCollection(DateHistogramAggregatorTests.java:979)
  2> NOTE: leaving temporary files on disk at: /home/brownbear/src/elasticsearch/server/build/testrun/test/temp/org.elasticsearch.search.aggregations.bucket.histogram.DateHistogramAggregatorTests_B0F2BA0C9CFDACD2-002
  2> NOTE: test params are: codec=Asserting(Lucene92): {_doc_count=Lucene90}, docValues:{f=DocValuesFormat(name=Lucene90)}, maxPointsInLeafNode=988, maxMBSortInHeap=7.320107332056045, sim=Asserting(RandomSimilarity(queryNorm=true): {}), locale=sr-BA, timezone=America/Thunder_Bay
  2> NOTE: Linux 5.15.0-33-generic amd64/Oracle Corporation 18.0.1.1 (64-bit)/cpus=48,threads=1,free=490950072,total=536870912
  2> NOTE: All tests run in this JVM: [DateHistogramAggregatorTests]

Tests with failures:
 - org.elasticsearch.search.aggregations.bucket.histogram.DateHistogramAggregatorTests.testTwoBucketsWithDocCountUsesTraditionalCollection

1 test completed, 1 failed

```

--- a/server/src/test/java/org/elasticsearch/search/aggregations/bucket/histogram/DateHistogramAggregatorTests.java
+++ b/server/src/test/java/org/elasticsearch/search/aggregations/bucket/histogram/DateHistogramAggregatorTests.java
@@ -54,6 +54,7 @@ import static org.elasticsearch.test.MapMatcher.assertMap;
 import static org.elasticsearch.test.MapMatcher.matchesMap;
 import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.greaterThanOrEqualTo;
 import static org.hamcrest.Matchers.hasSize;
 import static org.hamcrest.Matchers.instanceOf;
 import static org.hamcrest.Matchers.not;
@@ -997,7 +998,7 @@ public class DateHistogramAggregatorTests extends DateHistogramAggregatorTestCas
                                 "delegate_debug",
                                 matchesMap().entry("ranges", 2)
                                     .entry("average_docs_per_range", 5005.0)
-                                    .entry("singletons", greaterThan(1))
+                                    .entry("singletons", greaterThanOrEqualTo(1))
                                     .entry("non-singletons", 0)
                             )
                     )

